File: src/sys/dotnet/fan/sys/DirUriSpace.cs
//
// Copyright (c) 2008, Brian Frank and Andy Frank
// Licensed under the Academic Free License version 3.0
//
// History:
//   10 Mar 08  Andy Frank  Creation
//    9 Jul 09  Brian Frank Rename from DirNamespace
//

namespace Fan.Sys
{
  /// <summary>
  /// DirUriSpace aliases a directory tree
  /// </summary>
  internal sealed class DirUriSpace : UriSpace
  {

  //////////////////////////////////////////////////////////////////////////
  // Constructor
  //////////////////////////////////////////////////////////////////////////

    internal DirUriSpace(File dir)
    {
      if (!dir.isDir())  throw ArgErr.make("Not a dir: " + dir).val;
      if (!dir.exists()) throw ArgErr.make("Dir does not exist: " + dir).val;
      this.m_dir = dir;
    }

  //////////////////////////////////////////////////////////////////////////
  // Identity
  //////////////////////////////////////////////////////////////////////////

    public override string toStr() { return "DirUriSpace " + m_dir; }

    public override Type type() { return Sys.DirUriSpaceType; }

  //////////////////////////////////////////////////////////////////////////
  // UriSpace
  //////////////////////////////////////////////////////////////////////////

    public override object get(Uri uri, bool check)
    {
      File f = m_dir.plus(uri.relTo(this.uri()), false);
      if (f.exists()) return f;
      if (!check) return null;
      throw UnresolvedErr.make(uri).val;
    }

  //////////////////////////////////////////////////////////////////////////
  // Fields
  //////////////////////////////////////////////////////////////////////////

    private readonly File m_dir;

  }
}

File: src/sys/java/fan/sys/DirUriSpace.java
//
// Copyright (c) 2008, Brian Frank and Andy Frank
// Licensed under the Academic Free License version 3.0
//
// History:
//   04 Mar 08  Brian Frank  Creation
//    9 Jul 09  Brian        Rename from DirNamespace
//
package fan.sys;

import java.util.*;

/**
 * DirUriSpace aliases a directory tree
 */
final class DirUriSpace
  extends UriSpace
{

//////////////////////////////////////////////////////////////////////////
// Constructor
//////////////////////////////////////////////////////////////////////////

  DirUriSpace(File dir)
  {
    if (!dir.isDir())  throw ArgErr.make("Not a dir: " + dir).val;
    if (!dir.exists()) throw ArgErr.make("Dir does not exist: " + dir).val;
    this.dir = dir;
  }

//////////////////////////////////////////////////////////////////////////
// Identity
//////////////////////////////////////////////////////////////////////////

  public String toStr() { return "DirUriSpace " + dir; }

  public Type typeof() { return Sys.DirUriSpaceType; }

//////////////////////////////////////////////////////////////////////////
// UriSpace
//////////////////////////////////////////////////////////////////////////

  public Object get(Uri uri, boolean checked)
  {
    File f = dir.plus(uri.relTo(this.uri()), false);
    if (f.exists()) return f;
    if (!checked) return null;
    throw UnresolvedErr.make(uri).val;
  }

//////////////////////////////////////////////////////////////////////////
// Fields
//////////////////////////////////////////////////////////////////////////

  private final File dir;

}